lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -1131,6 +1131,10 @@ def message_from(response)
       def eligible_for_zero_auth?(payment_method, options = {})
         payment_method.is_a?(CreditCard) && options[:zero_amount_auth]
       end
+
+      def format_routing_number(routing_number, options)
+        options[:currency] == 'CAD' && routing_number.length > 8 ? routing_number[-8..-1] : routing_number
```

Comment:
Some special reason to use routing_number[-8..-1] instead number.slice!(0) ❓, personally I like more the rubiest way 😅

Reply:
well, two reasons:

1) the semantics of [-8, -1] mean "the latest 8 characters", while slice!(0) means removing the first element.
2) just for fun debate => Handling strings with ranges is more idiomatic in Ruby than slicing strings that you will see in the Js world.